code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -51,8 +51,18 @@ GLOBAL_LIST_INIT(blacklisted_malf_machines, typecacheof(list(
 	if(cooldown_period)
 		owner_AI.malf_cooldown = world.time + cooldown_period
 
+/datum/action/innate/ai/proc/update_desc()
+		desc = "[initial(desc)] There are [uses] reactivations remaining."
+		button.desc = desc
+
+/datum/action/innate/ai/New()
+	..()
+	if(initial(uses) > 1)
```

Comment:
You want to check for this inside update_desc instead, so that it only shows the uses after it updates if it has uses set.

Reply:
I thought so too, but like I said in the commit description this would make it so if you buy 2 fake alarms (which has an initial use of 1), and then use it once, the description wouldn't update and still say you had 2.

Reply:
If you look at the default value for uses in `/datum/action/innate/ai`, the vaule is null which means that you dont need to check if its greater than 1, just if it has been set to anything at all. Then this will work.

Reply:
That's a good catch, but it seems that all `one_purchase` powers without the `UPGRADE` variable or the `auto_use_uses` variable actually have it set so they have `uses = 1`. Removing this breaks most powers, including making the RCD power infinite

Reply:
Actually the doomsday device doesn't have a use count, so the button never goes away after you use it. It just has: 
```
if (active)
	return
```

Reply:
I see, in that case then I guess it doesn't matter too much